p2p/pex/pex_reactor.go

```diff
@@ -221,6 +221,7 @@ func (r *PEXReactor) Receive(chID byte, src Peer, msgBytes []byte) {
 		// 2) limit the output size
 		if r.config.SeedMode {
 			r.SendAddrs(src, r.book.GetSelectionWithBias(biasToSelectNewPeers))
+			src.FlushStop()
```

Comment:
so we're calling FlushStop() and after that Stop() in StopPeerGracefully()? Won't it make sense to have a Flush() func and leave stopping to OnStop?

pros:
- no code duplication
- separation of concerns

Reply:
The problem is, to ensure there's no races or new messages that get queued and not sent, we need the Flush func to stop the service so that all queued sends can be flushed and no new ones are concurrently created. I couldn't think of a better way to do it.

One alternative was to create a FlushSend method which waited until a send was actually flushed out on the conn, but that was more complicated than this solution.

Reply:
> there's no races or new messages that get queued and not sent, we need the Flush func to stop the service so that all queued sends can be flushed and no new ones are concurrently created.

Yeah, right. But why do you care about them. Here we only care about flushing one msg ^^^.

Reply:
Because we don't want to have a race condition in the code? We can't safely call `sendSomePacketMsgs` directly until the sendRoutine has exited otherwise we'll have a real race condition.

I agree with you in principle, but it would take some rearchitecting of the code to do what we want safely here